### PR TITLE
refactor: refactor CLI part of excel2xml (DEV-2190)

### PR DESCRIPTION
**Pull request by jnussbaum**
resolves DEV-2190

### CODE REVIEW

File: src/dsp_tools/excel2xml.py [hunk 1]
```diff
@@ -1806,152 +2037,207 @@ def excel2xml(datafile: str, shortcode: str, default_ontology: str) -> bool:
...
+        default_ontology: name of the ontology that this data belongs to
+
+    Raises:
+        BaseError if something went wrong
+
+    Returns:
+        True if everything went well, False otherwise
+    """
+    # read and prepare the input file
+    success = True
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply src/dsp_tools/excel2xml.py:2222]: it's modified on (new) line 2244

**Comment by jnussbaum** [comment]
> I like to be able to "read code from top to bottom" meaning that the public method should be art the beginning, and the private methods after that

Interesting, I had been thinking about this point, too. But I had always thought that the ideal would be the other way round: Define called functions above their caller. This is already the case in many DSP-TOOLS modules, and perhaps that makes sense for Python, because in "bare" Python code outside a `main()` function, a function cannot be called if it is defined further down.
It's also interesting what Bing's AI chat has to contribute to this question:
image
I'm especially struck by the last sentence: "Ideally, your modules should be small/short enough such that the ordering of the functions doesn’t really matter." Aouch, here I have some homework to do...

File: src/dsp_tools/excel2xml.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply src/dsp_tools/excel2xml.py:2222]: Ah... github stopped displaying me the file at line 2243 :)  
In that case I would suggest moving it down there, or even change the logic to something like this (pseudo code):

```python
if len(warnings) > 0:
    print("Finished with warnings")
    return False
print("finished...")
return True
```

or if you want to keep the flow you currently have

```python
with warnings ... as w:
    write(file)
    success: bool = len(w) == 0

print(finished...)
return success
```

**Comment by BalduinLandolt** [comment]
> > I like to be able to "read code from top to bottom" meaning that the public method should be art the beginning, and the private methods after that
> 
> Interesting, I had been thinking about this point, too. But I had always thought that the ideal would be the other way round: Define called functions above their caller. This is already the case in many DSP-TOOLS modules, and perhaps that makes sense for Python, because in "bare" Python code outside a `main()` function, a function cannot be called if it is defined further down. It's also interesting what Bing's AI chat has to contribute to this question: image I'm especially struck by the last sentence: "Ideally, your modules should be small/short enough such that the ordering of the functions doesn’t really matter." Aouch, here I have some homework to do...

Technically, it's not just outside a `main()` function, it has to be outside of any function at all. And this can never happen for DSP-TOOLS, I think.  
But I agree with the AI that it's a matter of taste, so if you prefer it that way, sure.

Regarding the last point, I mostly agree with the AI: modules should definitely be very short! (to me, 200-300 lines max seems sane in many cases, but it's hard to put a number on it. - This, by the way, is also why I'm not a fan of over-documenting code... this is a "Ballance-Akt" to document enough but not too much, otherwise code gets harder to read again.)  
But I'm also an advocate of many, short functions, which in turn makes ordering more important. So even for 200 lines of code, if you have short functions, it's very helpful to have a understandable ordering.

**Pull request merged**